### ISSUE TITLE
ECMA 6 to CoffeeScript 2.0 compatible version

### ISSUE THREAD

**Issue by auvipy** [js2coffee/js2coffee#475]

**Comment by anodynos** [comment]
Any thoughts on that? It would be really amazing now they are both at feature parity...

**Comment by auvipy** [comment]
i am not sure

**Comment by anodynos** [comment]
@rstacruz any chance you get on to it?

**Comment by rdeforest** [comment]
Issues https://github.com/jashkenas/coffeescript/issues/4969 and https://github.com/jashkenas/coffeescript/issues/4984 might address the need to translate the latest JS to the latest CoffeeScript.

(Long term I'm hoping the cs2js/js2cs projects will merge to permanently address this kind of problem.)

**Comment by owenstrevor** [comment]
This is sorely needed!

**Comment by AJeezy9** [comment]
Really wish this existed 🤕

**Comment by dimdenGD** [comment]
I don't see any reason to use js2coffee since it doesn't supports syntax that was created 4 years ago..

**Comment by jahbini** [comment]
I use it all the time.  Coffeescript is much more readable than es0 and far more readable than es6.

**Comment by dimdenGD** [comment]
> I use it all the time. Coffeescript is much more readable than es0 and far more readable than es6.

Welp, it's more readable for you, but not for people that are coding in normal JS.

**Comment by jahbini** [comment]
That is correct for the "write once, test ever, read never" coder on a team.  However, I need to qualify large numbers of projects and read massive amounts of code.  I can speed read in coffeescript to get the 30,000 level by reading the code.  I hope reading the code is not a lost skill.

**Comment by dimdenGD** [comment]
Also I don't have anything against CoffeeScript, I like it. Right now I'm talking about ES6 to CoffeeScript feature. It's really important, since not a lot of people using classes with function prototypes now, and other stuff from 2015. Cmon, it's been 4 years!

**Comment by jahbini** [comment]
Yes.  I hear polyfills and babel pipelines can be useful. The resultant universal JS can be used as a starting point.  Just think of it as a pretty printer for the syntactic strychnine of ES6. (Longbeard rant off)

**Comment by auvipy** [comment]
I might take on this myself in next 2/3 months.

**Comment by Kristinita** [comment]
@auvipy , is any progress?

Thanks.

**Comment by auvipy** [comment]
> @auvipy , is any progress?
> 
> Thanks.

things changed, I will revisit this again

**Comment by rdeforest** [comment]
Is there some way I can help?

**Comment by Kristinita** [comment]
This issue is still relevant as of December 2023.

Thanks.
